src/reflection.jl

```diff
@@ -0,0 +1,34 @@
+import InteractiveUtils
+export @ka_code_typed
+
+function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing)
+    # get the iterspace and dynamic of a kernel
+    ndrange, workgroupsize, iterspace, dynamic = KernelAbstractions.launch_config(kernel, ndrange, workgroupsize)
+    # get the first block
+    block = @inbounds KernelAbstractions.blocks(iterspace)[1]
+    # get a context of the kernel based on the first block
+    ctx = KernelAbstractions.mkcontext(kernel, block, ndrange, iterspace, dynamic)
+    # reformat
+    if argtypes isa Type
+        argtypes = argtypes.parameters
+    end
+    # use code_typed
+    return InteractiveUtils.code_typed(KernelAbstractions.Cassette.overdub, (typeof(ctx), typeof(kernel.f), argtypes...))
+end
+
+
+macro ka_code_typed(ex0...)
+    if length(ex0) == 1
+        kw = ex0[1].args[end]
```

Comment:
This only works for one argument? What about providing both `workgroupsize` and `ndrange`?

Reply:
If there is only one argument, it is because they called it with @ka_code_typed kernel(CPU(), ndrange=8) for example. They can also call it with @ka_code_typed ndrange=8 kernel(CPU()), but that might not be necessary to support.

Reply:
Right, but what about `@ka_code_typed kernel(args...; ndrange=128, workgroupsize=8)`?

Reply:
In that instance, I believe the whole kernel is the one argument ex0 that has different parameters inside of it. However, I think that I may need to split up in a similar way to get workgroup size to work for example